pkg/operator/ceph/cluster/nodedaemon/exporter.go

```diff
@@ -120,6 +120,7 @@ func (r *ReconcileNode) createOrUpdateCephExporter(node corev1.Node, tolerations
 			controller.AddCephVersionLabelToDeployment(*cephVersion, deploy)
 		}
 		var terminationGracePeriodSeconds int64 = 2
+		deploy.Spec.Strategy.Type = appsv1.RecreateDeploymentStrategyType
```

Comment:
What if there is just a container re-start does this handle that situation

Reply:
It doesn't handle that situation, that should be handled on the ceph-exporter layer

Reply:
@weirdwiz Has this solution been tested with the provider cluster to see if fixes the issue of socket file?

Reply:
yes, i've tested the install with a provider cluster
no longer crashes on install